extensions/kotlin-serialization/runtime/src/main/kotlin/io/quarkus/kotlin/serialization/JsonConfiguration.kt

```diff
@@ -0,0 +1,110 @@
+package io.quarkus.kotlin.serialization
+
+import io.quarkus.runtime.annotations.ConfigGroup
+import io.quarkus.runtime.annotations.ConfigItem
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.JsonNames
+
+@ConfigGroup
+open class JsonConfiguration {
+    /**
+     * Specifies whether default values of Kotlin properties should be encoded.
+     * `false` by default.
+     */
+    @ConfigItem(defaultValue = "false")
+    @JvmField
+    var encodeDefaults: Boolean = true
+    /**
+     * Specifies whether encounters of unknown properties in the input JSON
+     * should be ignored instead of throwing [SerializationException].
+     * `false` by default.
+     */
+    @ConfigItem(defaultValue = "false")
+    @JvmField
+    var ignoreUnknownKeys: Boolean = false
+    /**
+     * Removes JSON specification restriction (RFC-4627) and makes parser
+     * more liberal to the malformed input. In lenient mode quoted boolean literals,
+     * and unquoted string literals are allowed.
+     *
+     * Its relaxations can be expanded in the future, so that lenient parser becomes even more
+     * permissive to invalid value in the input, replacing them with defaults.
+     *
+     * `false` by default.
+     */
+    @ConfigItem(defaultValue = "false")
+    @JvmField
+    var isLenient: Boolean = false
+    /**
+     * Enables structured objects to be serialized as map keys by
+     * changing serialized form of the map from JSON object (key-value pairs) to flat array like `[k1, v1, k2, v2]`.
+     * `false` by default.
+     */
+    @ConfigItem(defaultValue = "false")
+    @JvmField
+    var allowStructuredMapKeys: Boolean = false
+    /**
+     * Specifies whether resulting JSON should be pretty-printed.
+     *  `false` by default.
+     */
+    @ConfigItem(defaultValue = "false")
+    @JvmField
+    var prettyPrint: Boolean = false
+    /**
+     * Specifies indent string to use with [prettyPrint] mode
+     * 4 spaces by default.
+     * Experimentality note: this API is experimental because
+     * it is not clear whether this option has compelling use-cases.
+     */
+    @ConfigItem(defaultValue = "    ")
+    @JvmField
+    var prettyPrintIndent: String = "    "
```

Comment:
You don't really need to specify the same value as the annotation. In Java we don't even do this, but I realize that in Kotlin you do need to set a value due to the nullability in the type system.
But maybe it makes more sense to allow these to be nullable? They will never be so in practice, but just to satisfy the type system... I think it would be best because having the default value in two places is confusing

Reply:
I've gone back and forth on that.  i would have preferred `lateinit` but that gets wonky with how quarkus processes the config classes.  leaving it like this makes the eventual configuration of the encoding simpler because I don't have to deal with the nullability.  I prefer this approach but I could be talked out of it.  how strongly do you feel about it?

Reply:
Not terribly strong - just a slight preference.

Reply:
But I am also slightly worried about what the tooling that builds the configuration docs will do in this case

Reply:
Is there any way I can generate those manually to check?  I have no idea how that part works.

Reply:
@machi1990 can you help out on ^?

Reply:
They are automatically generated in the project root `target/ascidoc/generated/config` folder when running `mvn clean install`. The file name you are looking will be `quarkus-kotlin-serialization.adoc`

Reply:
Thanks!

Reply:
the generated docs look ok to me.